src/AtelierTomato.Markov.Model/ObjectOID/InvalidObjectOID.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Text.RegularExpressions;
+
+namespace AtelierTomato.Markov.Model.ObjectOID
+{
+	public class InvalidObjectOID : IObjectOID
+	{
+		public ServiceType Service { get; } = ServiceType.Invalid;
+		public string Instance { get; } = "_";
+		public string Type { get; set; }
+		public InvalidObjectOID(string type)
+		{
+			if (type is "PermissionDenied")
+			{
+				Type = type;
+			}
+			else
+			{
+				throw new ArgumentException($"{type} is not a valid InvalidObjectOID Type.", nameof(type));

```

Comment:
Not a valid invalid... - you should rename this type and think about its role more.